visualizer/src/App.cpp

```diff
@@ -7,36 +7,61 @@
 using namespace mode7;
 
 static Shader test_shader;
-static Scene test_mesh;
+static Scene test_monk;
 static Scene test_cube;
-static float x;
-
+static float angle;
+static float radius; 
+static float moved; 
+static float x, y, z; 
 void App::init()
 {
     test_shader.open("assets/shaders/test_v.glsl", "assets/shaders/test_f.glsl");
-    test_mesh = ModelLoader::open("assets/models/suzanne.dae");
-    test_cube = ModelLoader::open("assets/models/test_cube.dae");
+    test_monk = ModelLoader::open("assets/models/suzanne.dae"); //Monkey object
+    //test_cube = ModelLoader::open("assets/models/test_cube.dae"); //Cube object
 
-    test_cube.position.z = -3.f;
+    test_monk.rotate(0.0f, 0.0f, 0.0f); //Rotate the object model to its front
+    test_cube.position = glm::vec3(0.0f, 0.0f, 0.0f); 
 
-    test_mesh.addChild(&test_cube);
+    radius = 2; //To move the object in a circle with radius 2 look at update method
+    angle = 0.0f; //Start at the beginning of the unit circle :D
+    x = y = z = 0.0f; 
+    //test_mesh.addChild(&test_cube); 
 }
 
+//Basically just focusing on the x,y plane as of now. 
 void App::update()
 {
-    test_cube.rotate(0.f, 0.01f, 0.f);
-    test_mesh.rotate(0.f, 0.01f, 0.f);
-    test_mesh.position.y = 3.f * sinf(x);
-    x += 0.05f;
+    //For object moving only up 
+    //test_cube.position.x = 0.0f; 
+    //test_cube.position.y = angle;
+ 
+    //For object moving in a circle. If the radius is larger, the object will go out of view at times so adjust the z-axis in the camera.
+    test_monk.position.x = radius * cosf(angle); 
+    test_monk.position.y = radius * sinf(angle);
+
+
+    //std::cout << angle << std::endl; 
 
-    test_mesh.update();
+    angle += 0.05f;  //Note since the screen's height is 480 --> an angle val larger than 4.8 takes the object out of view (a note for the commented code in beginning of this method)
+    test_monk.update(); 
+  
     
-    Camera::lookAt(glm::vec3(-5, 2, -5), test_mesh.position);
-    Camera::updateView();
+    //One view (circle movement)
+    Camera::lookAt(glm::vec3(test_cube.position.x, test_cube.position.y, 5), glm::vec3(0,0,0));
+
+    //Second view (circle movement)
+    //Camera::lookAt(glm::vec3(0.0f, 0.0f, 5), glm::vec3(0,0,0));
+   
+    //Try to change x, y, and z as the object is moving up (still figuring this out) to make the movement look seamless
+    //Camera::lookAt(glm::vec3(test_cube.position.x, test_cube.position.y, 5), glm::vec3(x, y, z));
+
+    //This will focus on the object, but it won't show any movement although the object is moving. 
+    //Camera::lookAt(glm::vec3(test_cube.position.x, test_cube.position.y, 5), test_cube.position);
```

Comment:
This doesn't show movement because the camera is always looking at test_cube.position, per the second argument to Camera::lookAt(camera_pos, object_pos). Test cube is moving but without a background, it is not visible.

Reply:
I believe you are looking at the first commit which has the wrong camera view code. The camera should be looking at the test_monk object as shown in the third view gif in my first comment. Should this be the main view as the object is moving in the 3D space (with keyboard functionality allowing for different camera views), or do you want to capture the movement as shown in view 1 or view 2? Or should I aim for something else? Like this [animation](https://www.youtube.com/watch?v=syKmJP3-Wec)? Or this [animation](https://www.youtube.com/watch?v=3zmMgR6HPDo)?

Reply:
**Something like view 3 is what is desired**. Keeping the vehicle in the center is probably easiest. There are ways to communicate acceleration/motion this way (Imagine the camera mounted to the vehicle on a critically damped spring, as the vehicle accelerates, it would pull away from the camera for a moment, before the camera catches up).